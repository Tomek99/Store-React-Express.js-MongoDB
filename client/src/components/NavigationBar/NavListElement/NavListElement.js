import React from "react";
import { NavLink, Link } from "react-router-dom";
import PropTypes from "prop-types";
import styles from "./NavListElement.module.scss";

function NavListElement({ isLink, name, path }) {
  return (
    <li className={styles.NavListElement}>
      {isLink ? (
        <Link to={path}>{name}</Link>
      ) : (
        <NavLink to={path} activeClassName={styles.activeTest}>
          {name}
        </NavLink>
      )}
    </li>
  );
}

NavListElement.propTypes = {
  isLink: PropTypes.bool,
  name: PropTypes.string,
  path: PropTypes.string,

  style: PropTypes.string,
};
export default NavListElement;
