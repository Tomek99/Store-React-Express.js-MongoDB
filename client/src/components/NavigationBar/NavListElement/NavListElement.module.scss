.NavListElement {
    .navLink {
        color: #fff;
        transition: 0.3s linear;
        position: relative;

        &::after {
            display: block;
            content: "";
            width: 0;
            height: 2px;
            background: var(--main-color);
            position: absolute;
            bottom: -3px;
            left: 0;
            transition: 0.3s linear;
        }

        &:hover::after {
            width: 100%;
        }

        &:hover {
            color: var(--main-color);
        }
    }

    .activeNavLink {
        color: var(--main-color);
        &::after {
            display: block;
            content: "";
            width: 100%;
            height: 2px;
            background: var(--main-color);
            position: absolute;
            bottom: -3px;
            left: 0;
            transition: 0.25s linear;
        }
    }
}
