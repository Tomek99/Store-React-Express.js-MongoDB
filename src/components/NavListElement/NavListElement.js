import React from "react";
import { NavLink, Link } from "react-router-dom";

function NavListElement(props) {
  const { isLink, name, path, activeStyle, style } = props;
  return (
    <li className={style} key={name}>
      {isLink ? (
        <Link to={path}>{name}</Link>
      ) : (
        <NavLink
          to={path}
          className={({ isActive }) => (isActive ? activeStyle : undefined)}
        >
          {name}
        </NavLink>
      )}
    </li>
  );
}

export default NavListElement;
