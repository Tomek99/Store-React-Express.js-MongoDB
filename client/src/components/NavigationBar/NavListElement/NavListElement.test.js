import { render } from "@testing-library/react";
import NavListElement from "./NavListElement";
import { MemoryRouter } from "react-router-dom";

test("should render NavListElement component", () => {
  const { container } = render(
    <MemoryRouter>
      <NavListElement isLink={true} name="test" path="/test" />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
