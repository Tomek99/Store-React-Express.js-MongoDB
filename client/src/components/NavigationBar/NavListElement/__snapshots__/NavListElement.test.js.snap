// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NavListElement component 1`] = `
<div>
  <li
    class="NavListElement"
  >
    <a
      href="/test"
    >
      test
    </a>
  </li>
</div>
`;
